internal/cli/main.go

```diff
@@ -190,8 +190,30 @@ Database drivers: `+strings.Join(database.List(), ", ")+"\n")
 			log.fatalErr(migraterErr)
 		}
 
-		limit := -1
-		if flag.Arg(1) != "" {
+		downFlagSet := flag.NewFlagSet("down", flag.ExitOnError)
+		applyAll := downFlagSet.Bool("all", false, "Apply all down migrations")
+
+		args := flag.Args()[1:]
+		if err := downFlagSet.Parse(args); err != nil {
+			log.Println(err)
+		}
+
+		limit := 0
+		if *applyAll {
+			limit = -1
+		} else if flag.Arg(1) == "" {
+			log.Println("Are you sure you want to apply all down migrations? [y/N]")
+			var response string
+			fmt.Scanln(&response)
+			response = strings.ToLower(strings.TrimSpace(response))
+
+			if response == "y" {
+				log.Println("Applying all down migrations")
+				limit = -1
+			} else {
+				log.Println("Not applying all down migrations")
+			}
+		} else {
```

Comment:
This doesn't quite work. The problem is that we get different behaviors if `down -all 5` is run vs `down 5 -all` since we're just looking at the first arg. I think we should error out if both `-all` and `N` are specified.

[FlagSet.Args()](https://golang.org/pkg/flag/#FlagSet.Args) should help. e.g. 
```golang
if (*applyAll) {
    limit = -1
} else {
    args := downFlagSet.Args()
    switch len(args) {
    case 0:
        // confirmation prompt
    case 1:
        // parse the arg as an int and run that many down migrations
    default:
        // error
    }
}
```

Reply:
what about this? With `down -all 5` applyAll is true, we would want to error instead of running all. If we want to error if there is more than one arg maybe have that as the first condition?

```golang
if len(args) > 1 {
  log.fatal("error: too many arguments")
} else if *applyAll {
  limit = -1
} else {
  downArgs := downFlagSet.Args()
  switch len(downArgs) {
  case 0:
    //
  case 1:
    //
  default:
    //
  }
}
```

Reply:
The docs state:
> Flag parsing stops just before the first non-flag argument ("-" is a non-flag argument) or after the terminator "--". 
> ...
> Args returns the non-flag arguments.

My interpretation of that is if `down -all 5` is specified, then `len(args)` would be `1`.
If `down 5 -all` is specified, then `len(args)` would be `2`.

Reply:
this snippet might help: https://play.golang.org/p/X59lUdA0Z1f

for `down -all 5`, `applyAll` is true and `len(downFlagSet.Args())` is 1
for `down 5 -all`, `applyAll` is false and `len(downFlagSet.Args())` is 2

so we get different behaviours for setting both

Reply:
Thanks for setting up the Go Playground link!
The output there matches my interpretation 😉 

So something like this should work:
```golang
downArgs := downFlagSet.Args()

if (*applyAll) { // handle flags before args
    if len(downArgs) > 1 {
        log.fatal("-all cannot be used with other arguments")
    }
    limit = -1
} else {
    switch len(downArgs) {
    case 0:
        // confirmation prompt
    case 1:
        // parse downArg[0] as an int and run that many down migrations
    default:
        // error
    }
}
```

Whatever logic we use, it should handle all of the cases. It may be worth moving this logic to a function (maybe named `numDownMigrationsFromArgs`) and adding table-driven tests since this wasn't as straightforward as we originally though.

Reply:
what do you think of the current version? now both `down -all N` and `down N -all` error